impl/src/main/java/org/jboss/weld/context/beanstore/AttributeBeanStore.java

```diff
@@ -160,7 +188,21 @@ protected NamingScheme getNamingScheme() {
     }
 
     public Iterator<BeanIdentifier> iterator() {
-        return beanStore.iterator();
+        Iterator<BeanIdentifier> iterator;
```

Comment:
> I'm not so sure. Is it possible to put a new bean instance in a bean store which is detached?

Not sure. My understanding of how AttributeBeanStores were originally designed was for them to be able to work detached and then sync with the backing store on attach. I am not sure however if this feature is actually used by any of the context implementations we use.